services/ui-src/src/utils/reports/entities.ts

```diff
@@ -15,6 +20,56 @@ const getCheckboxValues = (entity: EntityShape | undefined, label: string) => {
   );
 };
 
+const getReportingRateType = (entity: EntityShape | undefined) => {
+  return entity?.qualityMeasure_reportingRateType?.[0]?.value ===
+    "Cross-program rate"
+    ? "Cross-program rate: " +
+        entity?.qualityMeasure_crossProgramReportingRateProgramList
+    : entity?.qualityMeasure_reportingRateType?.[0]?.value;
+};
+
+const getReportingPeriod = (entity: EntityShape | undefined) => {
+  return entity?.qualityMeasure_reportingPeriod?.[0]?.value === "No"
+    ? `${entity?.qualityMeasure_reportingPeriodStartDate} - ${entity?.qualityMeasure_reportingPeriodEndDate}`
+    : entity?.qualityMeasure_reportingPeriod?.[0]?.value;
+};
+
+const getPlans = (entity: EntityShape, reportFieldData?: AnyObject) => {
+  // Check the entity to see how many plans are associated with the measure
+  const qualityMeasurePlanIdentifier = "qualityMeasure_plan_measureResults_";
+  const foundPlans = Object.keys(entity).filter((key) =>
+    key.includes(qualityMeasurePlanIdentifier)
+  );
+  let convertedPlans: AccessMeasureAssociatedPlan[] = [];
+
+  // Loop through all the plans that are found tied to the measure
+  for (let foundPlan of foundPlans) {
+    // We have the value already tied in from the users input on the field
+    const accessMeasureValue: string = entity[foundPlan];
+    /*
+     * We need the name of the plan however, and we're only given the ID at the end of the key.
+     * That means we need to look up in the reportFieldData for a matching key to know the plan name
+     */
+    const planName: string = reportFieldData?.plans?.find(
+      (plan: AnyObject) =>
+        plan.id === foundPlan.replace(qualityMeasurePlanIdentifier, "")
+    )?.name;
+
+    /*
+     * If we know the plan name and the value the user inputted for the access measure associated
+     * with the plan, then we can tie it together and add it to the list to return
+     */
+    if (planName && accessMeasureValue) {
+      const formattedAccessMeasure: AccessMeasureAssociatedPlan = {
+        name: planName,
+        value: accessMeasureValue,
+      };
+      convertedPlans.push(formattedAccessMeasure);
+    }
+  }
+  return convertedPlans;
+};
```

Comment:
Note: I don't think we're going to end up needing this anymore with the (likely) design changes to the card bottom. We'll render out a list of data sections, one for each plan, and display the plan name, then if there's associated data, display it. Something like this pseudocode:

```js
// for each plan in report data
reportDataPlans.forEach((plan: {id: string, name: string}) => {
    // prepare entered data here
    const enteredMeasureData = formattedEntityData.planMeasures;
    const planMeasureData = enteredMeasureData[`baselinestuff_plan_measure_${plan.id}`];
    // render a data section in the card bottom with plan name and data (if it exists)
    return (
        <DataSection>
            <Text>{plan.name}</Text>
            {planMeasureData && <Text>{planMeasureData}</Text>}
        </DataSection>
    )
})
```